LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -0,0 +1,1280 @@
+#define HDMICEC_METHOD_GET_CONNECTED_DEVICES "getConnectedDevices"
+#define HDMICEC_METHOD_SET_NAME "setName"
+#define HDMICEC_METHOD_SET_ONE_TOUCH_VIEW_POLICY "setOneTouchViewPolicy"
+
+#define HDMICEC_EVENT_ON_DEVICES_CHANGED "onDevicesChanged"
+#define HDMICEC_EVENT_ON_MESSAGE "onMessage"
+#define HDMICEC_EVENT_ON_HDMI_HOT_PLUG "onHdmiHotPlug"
+#define HDMICEC_EVENT_ON_CEC_ADDRESS_CHANGE "cecAddressesChanged"
+
+#define PHYSICAL_ADDR_CHANGED 1
+#define LOGICAL_ADDR_CHANGED 2
+#define DEV_TYPE_TUNER 1
+
+#define HDMI_HOT_PLUG_EVENT_CONNECTED 0
+#define HDMI_HOT_PLUG_EVENT_DISCONNECTED 1
+
+#if defined(HAS_PERSISTENT_IN_HDD)
+#define CEC_SETTING_ENABLED_FILE "/tmp/mnt/diska3/persistent/ds/cecData.json"
+#elif defined(HAS_PERSISTENT_IN_FLASH)
+#define CEC_SETTING_ENABLED_FILE "/opt/persistent/ds/cecData.json"
+#else
+#define CEC_SETTING_ENABLED_FILE "/opt/ds/cecData.json"
+#endif
+
+#define CEC_SETTING_ENABLED "cecEnabled"
+
+namespace
+{
+    using namespace WPEFramework;
+
+    inline void requestRescanning(int id)
+    {
+        LOGINFO("requestRescanning: id=%d", id);
+        IARM_Bus_CECHost_ConfigureScan_Param_t param  {};
+        param.needFullUpdate = true;
+        param.scanReasonId = id;
+
+        const IARM_Result_t ret = IARM_Bus_Call(IARM_BUS_CECHOST_NAME,
+                                                IARM_BUS_CEC_HOST_ConfigureScan,
+                                                (void *)&param,sizeof(param));
+
+        if (IARM_RESULT_SUCCESS != ret)
+        {
+            LOGERR("%s failed result %d", IARM_BUS_CEC_HOST_ConfigureScan, ret);
+        }
+    }
+    inline void setOsdName(const string& name)
+    {
+        IARM_Bus_CECHost_SetOSDName_Param_t param {};
+
+        strncpy((char *)param.name, name.c_str(), sizeof(param.name));
+        param.name[sizeof(param.name) - 1] = '\0';
+        const IARM_Result_t ret = IARM_Bus_Call(IARM_BUS_CECHOST_NAME,
+                                                IARM_BUS_CEC_HOST_SetOSDName,
+                                                (void *)&param,
+                                                sizeof(param));
+        if (IARM_RESULT_SUCCESS != ret)
+        {
+            LOGERR("%s failed result %d", IARM_BUS_CEC_HOST_SetOSDName, ret);
+        }
+    }
+}
+
+namespace WPEFramework
+{
+    namespace Plugin
+    {
+        SERVICE_REGISTRATION(LgiHdmiCec, 1, 0);
+
+        LgiHdmiCec* LgiHdmiCec::_instance = nullptr;
+
+        static std::atomic<int> libcecInitStatus{0};
+
+        LgiHdmiCec::LgiHdmiCec()
+        :   cecSettingEnabled(false),cecEnableStatus(false),smConnection(nullptr),
+            m_scan_id(0), m_updated(false), m_rescan_in_progress(true), m_system_audio_mode(false)
+        {
+            LgiHdmiCec::_instance = this;
+
+            InitializeIARM();
+            device::Manager::Initialize();
+
+            Register(HDMICEC_METHOD_SET_ENABLED, &LgiHdmiCec::setEnabledWrapper, this);
```

Comment:
Utils::Synchro::RegisterLockedApi should probably be used for method registration (see dt/helpers/UtilsSynchroIarm.hpp) - here and further on